client/src/components/Home/index.tsx

```diff
@@ -169,8 +184,39 @@ export default class Home extends React.Component<any, IHomeState> {
     this.setState({ infoVisible: false });
   }
 
+  onOpenImport() {
+    this.setState({ importVisible: true });
+  }
+
+  onCloseImport() {
+    this.setState({ importVisible: false });
+  }
+
+  updateFileName(value) {
+    this.setState({ fileName: value });
+  };
+
+  onLoad(importedFileContent, uploadResult) {
+    const { name, size, type, lastModifiedDate } = importedFileContent;
+    this.setState({ fileName: name.substr(0, name.indexOf('.')), content: uploadResult });
+  }
+
+  onSubmitImport() {
+    var dashboardId = this.state.fileName
+    ConfigurationsActions.submitDashboardFile(this.state.content, dashboardId);
+    
+    this.setState({ importVisible: false });
+  }
+
+  setFile(importedFileContent) {
+    this.setState({ importedFileContent });
+  }
+
   render() {
     let { loaded, redirectUrl, templates, selectedTemplateId, template } = this.state;
+    let { infoVisible, infoHtml } = this.state;
```

Comment:
This line needs to be removed as the `infoVisible` and `infoHtml` vars are redeclared on L220.